app/src/main/java/com/stevesoltys/seedvault/App.kt

```diff
@@ -143,8 +142,10 @@ fun <T> permitDiskReads(func: () -> T): T {
     }
 }
 
-fun Context.getSystemContext(isUsbStorage: () -> Boolean): Context {
-    return if (checkSelfPermission(INTERACT_ACROSS_USERS_FULL) == PERMISSION_GRANTED &&
-        isUsbStorage()
-    ) createContextAsUser(UserHandle.SYSTEM, 0) else this
+fun Context.getContext(isUsbStorage: () -> Boolean): Context {
```

Comment:
Maybe `getParentContext` ? Or something along that lines.

The main thing last time was to avoid confusion with the typically used `getContext` everywhere in Android, which this is not.

Reply:
This function takes a boolean as a parameter though. Otherwise, parent context would make it seem like we're always obtaining that context but that's not always applicable.

Reply:
My point here was it should be named something other than plain `getContext`, because this isn't that.

How about `getActualOrParentContext`